CMakeLists.txt

```diff
@@ -1,72 +1,252 @@
+#add_definitions( "-Wall -ansi -pedantic" )
+if(CMAKE_C_COMPILER_ID MATCHES "MSVC" OR CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+    # lots of warnings with cl.exe right now, use /W1
+    add_definitions("/W1 -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS /bigobj")
+else()
+    add_definitions("-Wall -O2 -Wno-unknown-pragmas -Wno-unused-function")
 endif()
 
-#######################################
+# install and export variables
+set(config_install_dir "lib/cmake/${PROJECT_NAME}")
+set(generated_dir "${CMAKE_CURRENT_BINARY_DIR}/generated")
+set(version_config "${generated_dir}/pydarknet-config-version.cmake")
+set(project_config "${generated_dir}/pydarknet-config.cmake")
+set(targets_export_name "pydarknet-targets")
+set(namespace "pydarknet::")
+
+if (SKBUILD)
+  # Override output paths when using scikit-build
+  set(PYDARKNET_LIB_INSTALL_DIR "pydarknet/lib")
+  set(PYDARKNET_INCLUDE_INSTALL_DIR "pydarknet/include")
+endif()
+
+if (NOT SKBUILD)
+  add_subdirectory( CMake )
+endif()
+add_subdirectory( src )
+if (BUILD_EXAMPLES)
+  add_subdirectory( examples )
+endif(BUILD_EXAMPLES)
+if (BUILD_TESTS)
+  add_subdirectory( test )
+endif (BUILD_TESTS)
 
-file(GLOB SOURCE_FILES
-    "src/*.c"
-    "src/*.h"
+# CMake configuration file creation
+# Include module with fuction 'write_basic_package_version_file'
+include(CMakePackageConfigHelpers)
+
+# Configure 'pydarknet-config-version.cmake'
+# Note: PYDARKNET_VERSION is used as a VERSION
+write_basic_package_version_file(
+    "${version_config}"
+    VERSION ${PYDARKNET_VERSION}
+    COMPATIBILITY SameMajorVersion
 )
 
-file(GLOB CUDA_SOURCE_FILES
-    "src/*.cu"
+# Configure 'pydarknet-config.cmake'
+# Use variables:
+#   * targets_export_name
+#   * PROJECT_NAME
+configure_package_config_file(
+  "cmake/Config.cmake.in"
+    "${project_config}"
+    INSTALL_DESTINATION "${config_install_dir}"
 )
 
-if (CUDA)
-    cuda_add_library(pydarknet.gpu SHARED
-        ${CUDA_SOURCE_FILES}
-    	${SOURCE_FILES}
-    )
+if (NOT SKBUILD)
+  # Config
+  #   * <prefix>/lib/cmake/pydarknet/pydarknet-config.cmake
+  #   * <prefix>/lib/cmake/pydarknet/pydarknet-config-version.cmake
+  install(
+      FILES "${project_config}" "${version_config}"
+      DESTINATION "${config_install_dir}"
+  )
+  # Config
+  #   * <prefix>/lib/cmake/pydarknet/pydarknet-targets.cmake
+  install(
+      EXPORT "${targets_export_name}"
+      NAMESPACE "${namespace}"
+      DESTINATION "${config_install_dir}"
+  )
+endif()
 
-    target_link_libraries(pydarknet.gpu
-        ${OpenCV_LIBS}
-        ${CUDA_LIBRARIES}
-        ${CMAKE_THREAD_LIBS_INIT}
-        ${CUDA_CUBLAS_LIBRARIES}
-        ${CUDA_curand_LIBRARY}
-    )
-else ()
-    add_library(pydarknet.cpu SHARED
-        ${SOURCE_FILES}
-    )
 
-    target_link_libraries(pydarknet.cpu
-        ${OpenCV_LIBS}
-        ${CUDA_LIBRARIES}
-        ${CMAKE_THREAD_LIBS_INIT}
-        ${CUDA_CUBLAS_LIBRARIES}
-        ${CUDA_curand_LIBRARY}
-    )
-endif ()
+# CPACK options
```

Comment:
Is there a strong reason to keep this cpack configuration around?

Reply:
In the event we want to host these outside of PyPI, CPack will help to greatly.  We can remove these down the line if we don't really intend to ever do that.  For now, it isn't hurting to keep this boilerplate code in place.